bundles/org.openhab.binding.energidataservice/README.md

```diff
@@ -47,15 +47,15 @@ It will not impact channels, see [Electricity Tax](#electricity-tax) for further
 
 ### Channel Group `electricity`
 
-| Channel                  | Type   | Description                                                                            | Advanced |
-|--------------------------|--------|----------------------------------------------------------------------------------------|----------|
-| spot-price               | Number | Current spot price in DKK or EUR per kWh                                               | no       |
-| grid-tariff              | Number | Current grid tariff in DKK per kWh. Only available when `gridCompanyGLN` is configured | no       |
-| system-tariff            | Number | Current system tariff in DKK per kWh                                                   | no       |
-| transmission-grid-tariff | Number | Current transmission grid tariff in DKK per kWh                                        | no       |
-| electricity-tax          | Number | Current electricity tax in DKK per kWh                                                 | no       |
-| reduced-electricity-tax  | Number | Current reduced electricity tax in DKK per kWh. For electric heating customers only    | no       |
-| hourly-prices            | String | JSON array with hourly prices from 24 hours ago and onward                             | yes      |
+| Channel                  | Type               | Description                                                                    | Advanced |
+|--------------------------|--------------------|--------------------------------------------------------------------------------|----------|
+| spot-price               | Number:EnergyPrice | Spot price in DKK or EUR per kWh                                               | no       |
+| grid-tariff              | Number:EnergyPrice | Grid tariff in DKK per kWh. Only available when `gridCompanyGLN` is configured | no       |
```

Comment:
You should probably add that - with the current implementations of `CurrencyProvider` - only one currency can be used.

Reply:
Good point, it seems I still don't fully understand even after your explanations in https://github.com/openhab/openhab-core/pull/3503#issuecomment-1849032580.

I ran some more extensive tests now:

| State              | Provider                    | Configuration | Result                                                                         |
|--------------------|-----------------------------|---------------|--------------------------------------------------------------------------------|
| 0.33126001 kr./kWh | LocaleBasedCurrencyProvider | Denmark       | 0.33126001 kr./kWh                                                             |
| 0.33126001 DKK/kWh | LocaleBasedCurrencyProvider | Denmark       | java.lang.IllegalArgumentException: Invalid Quantity value: 0.33126001 DKK/kWh |
| 0.33126001 kr./kWh | LocaleBasedCurrencyProvider | Sweden        | java.lang.IllegalArgumentException: Invalid Quantity value: 0.33126001 kr./kWh |
| 0.33126001 kr./kWh | FixedCurrencyProvider       | DKK           | 0.33126001 kr./kWh                                                             |
| 0.33126001 kr./kWh | FixedCurrencyProvider       | EUR           | 0.33126001 €./kWh                                                              |
| 0.33126001 DKK/kWh | FixedCurrencyProvider       | DKK           | java.lang.IllegalArgumentException: Invalid Quantity value: 0.33126001 DKK/kWh |
| 0.044770 €/kWh     | FixedCurrencyProvider       | DKK           | java.lang.IllegalArgumentException: Invalid Quantity value: 0.044770 €/kWh     |
| 0.044770 €/kWh     | FixedCurrencyProvider       | EUR           | 0.044770 €/kWh                                                                 |

I have one channel which can be configured as either DKK or EUR, and all other channels are DKK. I guess this cannot work with UoM because it's not possible mix currencies? Also it seems that it's not possible to use any currency other than the configured one, although I'm a bit confused why case 3 and 5 have different results?

Perhaps you can advise how to avoid `IllegalArgumentException`? Can the binding check something before trying to update state with a specific currency?

Reply:
@J-N-K - I don't know if you are receiving notifications without being tagged explicitly, so just to be sure, please see above when you find time.